worker/src/main/java/gov/cms/ab2d/worker/processor/PatientContractCallable.java

```diff
@@ -5,63 +5,105 @@
         try {
             ContractMapping mapping = new ContractMapping();
             mapping.setMonth(month);
             Bundle bundle = getBundle(contractNumber, month);
-            final Set<String> patientIDs = extractPatientIDs(bundle);
+            patientIds.addAll(extractAndFilter(bundle));
 
             while (bundle.getLink(Bundle.LINK_NEXT) != null) {
                 bundle = bfdClient.requestNextBundleFromServer(bundle);
-                patientIDs.addAll(extractPatientIDs(bundle));
+                patientIds.addAll(extractAndFilter(bundle));
             }
-            mapping.setPatients(patientIDs);
-            log.debug("finished reading [{}] Set<String>resources", patientIDs.size());
+            mapping.setPatients(patientIds);
+            log.debug("finished reading [{}] Set<String>resources", patientIds.size());
             return mapping;
         } catch (Exception e) {
             log.error("Unable to get patient information for " + contractNumber + " for month " + month, e);
             throw e;
+        } finally {
+
+            int total = patientIds.size() + pastYear + missingIdentifier;
+            log.info("Search discarded {} entries not meeting year filter criteria out of {}", pastYear, total);
+            log.info("Search discarded {} entries missing an identifier out of {}", missingIdentifier, total);
+
         }
     }
 
-    /**
-     * Given a Bundle, filters resources of type Patient and returns a list of patientIds
-     *
-     * @param bundle the bundle to extract data from
-     * @return a list of patientIds
-     */
-    private Set<String> extractPatientIDs(Bundle bundle) {
-        return bundle.getEntry().stream()
-                .map(Bundle.BundleEntryComponent::getResource)
-                .filter(resource -> resource.getResourceType() == ResourceType.Patient)
-                .map(resource -> (Patient) resource)
+    private Set<String> extractAndFilter(Bundle bundle) {
+        return getPatientStream(bundle)
+                .filter(this::filterByYear)
                 .map(this::extractPatientId)
-                .filter(StringUtils::isNotBlank)
-                .collect(Collectors.toSet());
+                .filter(this::notMissingIdentifier)
+                .collect(toSet());
+    }
+
+    private Stream<Patient> getPatientStream(Bundle bundle) {
+        return bundle.getEntry().stream().map(Bundle.BundleEntryComponent::getResource)
+                .filter(resource -> resource.getResourceType() == ResourceType.Patient)
+                .map(resource -> (Patient) resource);
+    }
+
+    private boolean filterByYear(Patient patient) {
+        List<Extension> referenceYearList = patient.getExtensionsByUrl("https://bluebutton.cms.gov/resources/variables/rfrnc_yr");
+
+        if (referenceYearList.isEmpty()) {
+            log.error("patient returned without reference year violating assumptions");
+            pastYear += 1;
+            return false;
+        } else {
```

Comment:
No need for _} else {_ since there is a return before.  Helps keep cyclomatic complexity down.